### PR TITLE
Dylan and Advaith / Snippet Functionality 

### LINKED ISSUE #54: Implement snippet functionality
for the snippets box on the dashboard, implement functionality such that:
- you can create and post code snippets
   - include a textbox to type in a code snippet
   - include a dropdown to select its coding language
   - include a submit button to post the snippet 
- interact with the database such that the snippet text, language, and date are saved
- code snippets should be displayed with the most recent addition at the top
- code snippets are unique to the day its posted

### PR DESCRIPTION
**Pull request by dylantrann**
## Tracking Info

Resolves #54

## Changes

index.html
- added `<div id="snippets"> `
   - includes all HTML elements involved in the snippets section of the dashboard
- added `<div id="alert">`
   - used for alerts when copying an existing snippet or adding new snippets

snippetCreation/snippetCreation.css
- uses #25 for general frame of snippets section
- contains all CSS styling for snippets section
- also contains CSS for alert window
- TODO: adding snippets and alert window were not included in design; might need to consult design lead

snippetCreation/snippetCreation.js
- database functionality including uploading and retrieving
- creates HTML objects to display retrieved snippets
- able to copy code snippet to clipboard
- pop up to alert when snippet successfully added or copied

## Testing

Used Docker and tested on local server.
- Checked that database was properly updated in Docker
- Checked that displayed correctly on local server

## Confirmation of Change

image

image

image

image

### CODE REVIEW

File: source/snippetCreation/snippetCreation.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// event listener to handle DOM Content load first upon which requests can be made
+document.addEventListener('DOMContentLoaded', () => {
+    const snippetForm = document.getElementById('snippet-form');
+    snippetForm.addEventListener('submit', (event) => {
```
Comment (EdwardLinS) [inline source/snippetCreation/snippetCreation.js:4]: When the selecting date is implemented with showing snippets and a snippet is added, if the user isn't on the same date, they don't see any changes to indicate a snippet being added which is correct but hard for users to know if they submitted anything. 

Is it possible to change the button text to text like "Snippet Added!" for a few seconds?

**Review by EdwardLinS — CHANGES_REQUESTED** [review]
The functionalities for fetching and adding snippets work well and it matches Figma design! The only request is to add an indicator for adding snippets. Other than that change, great job!

File: source/snippetCreation/snippetCreation.js (hunk 1 above)
Comment (dylantrann) [reply source/snippetCreation/snippetCreation.js:4]: working on it rn

Comment (dylantrann) [reply source/snippetCreation/snippetCreation.js:4]: added the new functionality. should have an alert pop up for both adding new snippets and copying existing snippets

**Pull request merged**